src/test/java/step4/AppTest.java

```diff
@@ -0,0 +1,79 @@
+package step4;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class AppTest {
+
+	private InputConsole inputConsole;
+
+	@BeforeEach
+	void init() {
+		inputConsole = new InputConsole();
+	}
+
+	@DisplayName("자동차가 4 이상일 때, 가는지 결과를 확인하는 테스트")
+	@Test
+	void goTest() {
+		Car car = new Car("yul", 0);
+		car.move(new NumberGenerator() {
+			@Override
+			public int getNumber() {
+				return 4;
+			}
+		});
```

Comment:
```suggestion
		car.move(()->4);
```
Functional Interface 를 활용해보아요.